elasticdl/python/elasticdl/layers/embedding.py

```diff
@@ -205,7 +205,10 @@ def call(self, input):
         outputs = tf.gather(batch_embedding, idx)
         # tf.reshape does not support shape with None. Replace None with -1.
         if ids.get_shape().rank == 2:
-            output_shape = (-1, ids.get_shape()[1], self.output_dim)
+            input_dim = (
+                -1 if ids.get_shape()[1] is None else ids.get_shape()[1]
+            )
+            output_shape = (-1, input_dim, self.output_dim)
```

Comment:
TensorFlow does not support shape containing two `-1` for reshape.

Reply:
yeah, I have fixed the default value to `1`.

Reply:
I think input_dim in Embedding Layer means the size of size of the vocabulary. It's used to make a constraint that the id shouldn't get out of input_dim, other wise it's out of range.
ids.get_shape()[1] means the input_length

Reply:
And I'm wondering what's the scenario that the input_length(aka ids.get_shape()[1]) will be none

Reply:
I think I have mixed the input_dim and input_length. I will rename the input_dim to input_length like:
```python
input_length = (
 1 if ids.get_shape()[1] is None else ids.get_shape()[1]
)
output_shape = (-1, input_length, self.output_dim)
```

Reply:
When we use tf.keras.model.clone_model to clone a new sequential model with elasticdl.Embedding, the`clone_model` will call `call` of elasticdl.Embedding with inputs And the shape of inputs is (None, None).

Reply:
> When we use tf.keras.model.clone_model to clone a new sequential model with elasticdl.Embedding, the`clone_model` will call `call` of elasticdl.Embedding with inputs And the shape of inputs is (None, None).

When we are building the Squential model with native embedding layer, it invoke  keras.layers.embedding.call with input(None, None) at the first time. The out is a tensor with shape (None, None, output_dim) without any value inside. Could we keep the same behavior and don't do the reshape?

Reply:
Besides the `clone_model`, input(None, None) is supported in Keras to build model like the demo:
```
inputs = Input(shape=(None,))
embedding = Embedding(4,2)(inputs)
model = tf.keras.Model(inputs=[inputs], outputs=[embedding])
```

Do you mean that we return the outputs without reshaping when ids.get_shape()[1] is None?

Reply:
I think input (None, None) is not for clone model. It's also invoked in the Squential Api. I think keras use this way to build the layer and initialize the variables. Not for real calculation.

Reply:
I agree with it. So we also should support building the elasticdl.Embedding layer with input(None, None).